test/runtests.jl

```diff
@@ -191,6 +192,39 @@ end
     @test @inferred(f9(plan_fft(zeros(10), 1), 10)) == 1/10
 end
 
+@testset "adjoint" begin
+    @testset "complex fft adjoint" begin
+        for x in (randn(3), randn(3, 4), randn(3, 4, 5))
+            N = ndims(x)
+            y = randn(size(x))
+            for dims in unique((1, 1:N, N))
+                P = plan_fft(x, dims)
+                @test dot(y, P * x) ≈ dot(P' * y, x)
+                @test_broken dot(y, P \ x) ≈ dot(P' \ y, x)
+                Pinv = plan_ifft(x)
+                @test dot(x, Pinv * y) ≈ dot(Pinv' * x, y)
+                @test_broken dot(x, Pinv \ y) ≈ dot(Pinv' \ x, y)
+            end
+        end
+    end
+    @testset "real fft adjoint" begin
+        for x in (randn(3), randn(4), randn(3, 4), randn(3, 4, 5)) # test odd and even lengths
+            N = ndims(x)
+            for dims in unique((1, 1:N, N))
+                P = plan_rfft(similar(x), dims)        
+                y_real = randn(size(P * x))
+                y_imag = randn(size(P * x))
+                y = y_real .+ y_imag .* im 
+                @test dot(y_real, real.(P * x)) + dot(y_imag, imag.(P * x)) ≈ dot(P' * y, x)
```

Comment:
for the complex FFT what's been implemented is clearly the adjoint, not the transpose. For the real FFT it's a little less clear, as on one hand we still take the adjoint of `F` but on the other hand, as this line shows, since it seems like the right dot product to use for the outputted complex vector is to roll it out into a real vector. Adjoint still seems like a safe term though